src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs

```diff
@@ -26,5 +28,44 @@ private static bool GetGlobalizationInvariantMode()
             }
             return invariantEnabled;
         }
+
+        private static bool LoadIcu()
+        {
+            if (!TryGetAppLocalIcuSwitchValue(out ReadOnlySpan<char> version, out ReadOnlySpan<char> suffix))
+            {
+                return Interop.Globalization.LoadICU() != 0;
+            }
+
+#if TARGET_OSX
+            string extension = ".dylib";
+            bool versionAtEnd = false;
+#else
+            string extension = "so.";
+            bool versionAtEnd = true;
+#endif
+
+            // Append '.' to suffix since in Unix the version is separated by '.'
+            int suffixLength = suffix.Length + 1;
+            Span<char> suffixWithSeparator = stackalloc char[suffixLength];
+            suffix.CopyTo(suffixWithSeparator);
+            suffixWithSeparator[suffixLength - 1] = '.';
+
+            Assembly assembly = Assembly.GetExecutingAssembly();
+
+#if !TARGET_OSX
+            // In Linux we need to load libicudata first because libicuuc and libicui18n depend on it. In order for the loader to find
```

Comment:
This sounds very odd, but there are soo many odd things about the ICU packaging and loading that I can believe it.

Reply:
I think this complication is coming from wanting to be able to place the ICU libraries in an arbitrary location that the OS loader doesn't know anything about (and the path isn't set in any env variable) -- but I think your point still stands regardless. :)

Reply:
Right, this is because we're trying to load from an arbitrary location that the OS loader might not know about if the environment variables to tell the OS loader, consider this path are not set.